src/transformStateWithClones.js

```diff
@@ -7,7 +7,25 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const clonesOfState = [];
+  const stateCopy = { ...state };
+
+  for (let i = 0; i < actions.length; i++) {
+    if (actions[i].type === 'addProperties') {
+      Object.assign(stateCopy, actions[i].extraData);
+    } else if (actions[i].type === 'removeProperties') {
+      for (const key of actions[i].keysToRemove) {
+        delete stateCopy[key];
+      }
+    } else if (actions[i].type === 'clear') {
+      for (const member in stateCopy) {
```

Comment:
```suggestion
      for (const member in stateCopy) {
```
member is a strange name for this variable